02_Util/src/authorizer/RealTimeAuthorizer.ts

```diff
@@ -0,0 +1,101 @@
+// Copyright 2025 Contributors to the CitrineOS Project
+//
+// SPDX-License-Identifier: Apache 2.0
+import { Authorization, ILocationRepository } from '@citrineos/data';
+import {
+  AuthorizationStatusEnumType,
+  IAuthorizationDto,
+  IAuthorizer,
+  IMessageContext,
+  RealTimeAuthEnumType,
+} from '@citrineos/base';
+import { ILogObj, Logger } from 'tslog';
+import { auth } from '@directus/sdk';
+
+export class RealTimeAuthorizer implements IAuthorizer {
+  private _locationRepository: ILocationRepository;
+  private readonly _logger: Logger<ILogObj>;
+
+  constructor(locationRepository: ILocationRepository, logger?: Logger<ILogObj>) {
+    this._locationRepository = locationRepository;
+    this._logger = logger
+      ? logger.getSubLogger({ name: this.constructor.name })
+      : new Logger<ILogObj>({ name: this.constructor.name });
+  }
+
+  async authorize(
+    authorization: Authorization,
+    context: IMessageContext,
+  ): Promise<AuthorizationStatusEnumType> {
+    if (!authorization.realTimeAuthUrl) {
+      this._logger.debug(`No Realtime Auth URL from authorization ${authorization.id}`);
+      return authorization.status;
+    } else if (
+      !authorization.realTimeAuth ||
+      authorization.realTimeAuth === RealTimeAuthEnumType.Allowed
+    ) {
```

Comment:
When realtime is allowed, why we skip the realtime auth call and return? I might misunderstand the meaning this `RealTimeAuthEnumType.Allowed` or it might be a typo of `AllowedOffline`?